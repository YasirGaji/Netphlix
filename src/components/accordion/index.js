import React, { useState, useContext, createContext } from 'react'
import {
  Container,
  Inner,
  Title,
  Item,
  Header,
  Body,
  Frame,
} from './styles/accordion';

const ToggleContext = createContext();

export default function Accordion({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  )
}

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>
}

// Accordion.Item = function AccordionItem({ children, ...restProps }) {
//   const [toggleShow, setToggleShow] = useState(false)

//   return (
//     <Item {...restProps}>
//       <Header onClick={() => setToggleShow((toggleShow) => !toggleShow)}>
//         {children}
//       </Header>
//       <Body className={toggleShow ? 'open' : 'close'}>{children}</Body>
//     </Item>
//   )
// }

Accordion.Item = function AccordionItem({ children, ...restProps }) {
  const [toggleShow, setToggleShow] = useState(false);

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  )
}


Accordion.Header = function AccordionHeader({ children, ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { toggleShow, setToggleShow } = useContext(ToggleContext);

  return (
    <Header onClick={() => setToggleShow((toggleShow) => !toggleShow)} {...restProps}>
      {children}
    </Header>
  )
}


Accordion.Body = function AccordionBody({ children, ...restProps }) {
  return <Body {...restProps}>{children}</Body>
}
