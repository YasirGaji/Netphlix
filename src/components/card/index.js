import React from 'react'
import {
  Container,
  Group,
  Title,
  SubTitle,
  Text,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Maturity,
  Content,
  Meta,
  Entities,
  Item,
  Image,
} from './styles/card'

export default function Card({ children, ...restProps}) {
  return <Container { ...restProps}>{children}</Container>
}

Card.Group = function CardGroup({ children, ...restProps}) {
  return <Group { ...restProps}>{children}</Group>
}

Card.Title = function CardTitle({ children, ...restProps}) {
  return <Title { ...restProps}>{children}</Title>
}

Card.SubTitle = function CardSubTitle({ children, ...restProps}) {
  return <SubTitle { ...restProps}>{children}</SubTitle>
}

Card.Text = function CardText({ children, ...restProps}) {
  return <Text { ...restProps}>{children}</Text>
}

Card.Feature = function CardFeature({ children, category, ...restProps}) {
  return <Feature { ...restProps} src={`/images/${category}/${category}.png`}>{children}</Feature>
}

Card.FeatureTitle = function CardFeatureTitle({ children, ...restProps}) {
  return <FeatureTitle { ...restProps}>{children}</FeatureTitle>
}

Card.FeatureText = function CardFeatureText({ children, ...restProps}) {
  return <FeatureText { ...restProps}>{children}</FeatureText>
}

Card.FeatureClose = function CardFeatureClose({ children, ...restProps}) {
  return <FeatureClose { ...restProps}>{children}</FeatureClose>
}

Card.Maturity = function CardMaturity({ children, ...restProps}) {
  return <Maturity { ...restProps}>{children}</Maturity>
}

Card.Content = function CardContent({ children, ...restProps}) {
  return <Content { ...restProps}>{children}</Content>
}

Card.Meta = function CardMeta({ children, ...restProps}) {
  return <Meta { ...restProps}>{children}</Meta>
}

Card.Entities = function CardEntities({ children, ...restProps}) {
  return <Entities { ...restProps}>{children}</Entities>
}

Card.Item = function CardItem({ item, children, ...restProps}) {
  const { genre } = item
  return (
    <Item { ...restProps}>
      <Image src={`/images/${genre}/${item.slug}/small.jpg`} />
      {children}
    </Item>
  )
}

Card.Image = function CardImage({ ...restProps}) {
  return <Image { ...restProps} />
}

