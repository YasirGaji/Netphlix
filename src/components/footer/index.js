import React from 'react'
import { 
  Container,
  Row,
  Column,
  Link,
  Title,
  Text,
  Break
} from './styles/footer'

export default function Footer({ children, ...restProps}) {
  return (
    <Container {...restProps}>{children}</Container>
  )
}


Footer.Row = function FooterRow({ children, ...restProps}) {
  return <Row {...restProps}>{children}</Row>
}

Footer.Column = function FooterRow({ children, ...restProps}) {
  return <Column {...restProps}>{children}</Column>
}

Footer.Link = function FooterRow({ children, ...restProps}) {
  return <Link {...restProps}>{children}</Link>
}

Footer.Title = function FooterRow({ children, ...restProps}) {
  return <Title {...restProps}>{children}</Title>
}

Footer.Text = function FooterRow({ children, ...restProps}) {
  return <Text {...restProps}>{children}</Text>
}

Footer.Break = function FooterRow({ children, ...restProps}) {
  return <Break {...restProps}>{children}</Break>
}