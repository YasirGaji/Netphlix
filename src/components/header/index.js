import React from 'react'
import { 
  Background, 
  Container, 
  Logo, 
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Search,
  SearchIcon,
  SearchInput,
  Picture,
  Profile,
  Dropdown,
} from './styles/header'
import { Link as ReactRouterLink } from 'react-router-dom'


export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
}

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
}

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
  const [searchActive, setSearchActive] = React.useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setSearchActive(searchActive => !searchActive)}>
        <img src="/images/icons/search.png" alt="Search" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search films and series by yasir"
        active={searchActive}
      />
    </Search>
  );
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>; 
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
}