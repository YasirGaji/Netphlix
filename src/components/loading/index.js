import React from "react";
import { Container, Picture } from "./styles/loading";


export const Loading = ({ src, ...restProps }) => {
  return (
    <Container {...restProps}>
      <Picture src={`/images/users/${src}.png`} />
    </Container>
  );
}

