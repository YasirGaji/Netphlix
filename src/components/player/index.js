import React, { useState, useContext, createContext } from "react";


