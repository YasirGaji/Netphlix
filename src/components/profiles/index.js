import React from "react";
import { 
  Container, 
  Title, 
  List, 
  User, 
  Picture,
  Name 
} from "./styles/profiles";

export default function Profiles({children, ...restProps}) {
  return <Container {...restProps}>{children}</Container>
}

Profiles.Title = function ProfilesTitle({children, ...restProps}) {
  return <Title {...restProps}>{children}</Title>
}

Profiles.List = function ProfilesList({children, ...restProps}) {
  return <List {...restProps}>{children}</List>
}

Profiles.User = function ProfilesUser({children, ...restProps}) {
  return <User {...restProps}>{children}</User>
}

Profiles.Picture = function ProfilesPicture({src, ...restProps}) {
  return <Picture {...restProps} src={src ? `/images/users/${src}.png` : `/images/misc/loading.gif`} />
}

Profiles.Name = function ProfilesName({children, ...restProps}) {
  return <Name {...restProps}>{children}</Name>
}

