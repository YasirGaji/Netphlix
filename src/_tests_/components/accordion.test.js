/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/prefer-presence-queries */
/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Accordion } from "../../components";
import faqsData from "../../fixtures/faqs.json";

describe("<Accordion />", () => {
  it("renders the <Accordion /> with populated data", () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Frame>
          {faqsData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Frame>
      </Accordion>
    );

    expect(getByText("Frequently Asked Questions")).toBeTruthy();
    expect(getByText("What is Netphlix?")).toBeTruthy();
    expect(getByText("How much does Netphlix cost?")).toBeTruthy();
    expect(getByText("Where can I watch?")).toBeTruthy();
    expect(getByText("How do I cancel?")).toBeTruthy();
    expect(getByText("What can I watch on Netphlix?")).toBeTruthy();
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });

  // it("opens and closes the <Accordion /> component", () => {
  //   const { container, queryByText } = render(
  //     <Accordion>
  //       <Accordion.Title>Frequently Asked Questions</Accordion.Title>
  //       <Accordion.Frame>
  //         {faqsData.map((item) => (
  //           <Accordion.Item key={item.id}>
  //             <Accordion.Header>{item.header}</Accordion.Header>
  //             <Accordion.Body>{item.body}</Accordion.Body>
  //           </Accordion.Item>
  //         ))}
  //       </Accordion.Frame>
  //     </Accordion>
  //   );

  //   expect(queryByText("Netphlix is a streaming service that offers a wide variety of award-winning TV shows, movies, anime, documentaries, and more on thousands of internet-connected devices.")).toBeFalsy();
  //   fireEvent.click(queryByText("What is Netphlix?"));
  //   expect(queryByText("Netphlix is a streaming service that offers a wide variety of award-winning TV shows, movies, anime, documentaries, and more on thousands of internet-connected devices.")).toBeTruthy();
  //   fireEvent.click(queryByText("What is Netphlix?"));
  //   expect(queryByText("Netphlix is a streaming service that offers a wide variety of award-winning TV shows, movies, anime, documentaries, and more on thousands of internet-connected devices.")).toBeFalsy();
  //   expect(container.firstChild).toMatchSnapshot();
  // });

})