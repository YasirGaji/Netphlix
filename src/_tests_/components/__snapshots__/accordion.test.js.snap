// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-sc-chm09z-0 dOXMfr"
>
  <div
    class="accordion__Inner-sc-chm09z-2 kRBjCz"
  >
    <h1
      class="accordion__Title-sc-chm09z-3 gUJWrA"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Frame-sc-chm09z-1 hdlQuW"
    >
      <div
        class="accordion__Item-sc-chm09z-4 jiYZjE"
      >
        <div
          class="accordion__Header-sc-chm09z-5 dWANDq"
        >
          What is Netphlix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-chm09z-4 jiYZjE"
      >
        <div
          class="accordion__Header-sc-chm09z-5 dWANDq"
        >
          How much does Netphlix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-chm09z-4 jiYZjE"
      >
        <div
          class="accordion__Header-sc-chm09z-5 dWANDq"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-chm09z-4 jiYZjE"
      >
        <div
          class="accordion__Header-sc-chm09z-5 dWANDq"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-chm09z-4 jiYZjE"
      >
        <div
          class="accordion__Header-sc-chm09z-5 dWANDq"
        >
          What can I watch on Netphlix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
